architecture/adr-list/adr-002-did-linked-resources.md

```diff
@@ -0,0 +1,834 @@
+        "resourceId": "44547089-170b-4f5a-bcbc-06e46e0089e4", // Old Resource ID and version number
+        "resourceName": "PassportSchema", // Resource name must remain the same
+        "resourceType": "CL-Schema", // Resource type must remain the same
+        "resourceVersion": "1.0.1", // A user-set version
+        "mediaType": "application/json",
+        "created": "2022-07-19T08:40:00Z",
+        "checksum": "7b2022636f6e74656e74223a202274657374206461746122207d0ae3b0c44298", // Old version checksum
+        "previousVersionId": null, // null, since no previous version
+        "nextVersionId": "bb2118f3-5e55-4510-b420-33ef9e1726d2", // Points to next version below
+        },
+      { // Second version of a Resource called PassportSchema
+        "resourceURI": "did:cheqd:testnet:13d5ad44-9e99-428f-81e9-274458cefddc/resources/bb2118f3-5e55-4510-b420-33ef9e1726d2",
+        "resourceCollectionId": "DAzMQo4MDMxCjgwM", // Common collection ID
+        "resourceId": "bb2118f3-5e55-4510-b420-33ef9e1726d2", // New Resource ID and version number
+        "resourceName": "PassportSchema", // Resource name must remain the same
+        "resourceType": "CL-Schema", // Resource type must remain the same
+        "resourceVersion" "1.0.1" // user-set semantic version control
+        "mediaType": "application/json",
+        "created": "2022-08-07T08:40:00Z",
+        "checksum": "9123dcbb0b42652b0e105956c68d3ca2ff34584f324fa41a29aedd32b883e131", // New version checksum
+        "previousVersionId": "44547089-170b-4f5a-bcbc-06e46e0089e4", // Points to previous version above
+        "nextVersionId": "0be87654-4a48-4f8e-8789-15ec3589ccdd" // Points to next version. Empty string if no new version
+        }
+      ]
+    }
+```
+
+Importantly, we decided not to populate the actual resource data into the didDocumentMetadata, but instead, what we refer to as a _Resource Preview_ which contains all the metadata about the associated resources.
+
+### Resource Preview
+
+Resource previews will aopear within DIDDocMetadata. These do not include the actual core data of the resource and only reference the metadata: 
+
+* Resource Collection ID: (did:cheqd:...:) (supplied client-side)\*\*
+* Resource ID: UUID ➝ specific to resource, also effectively a version number (supplied client-side)
+* Resource Name: String (e.g., `CL-Schema1` (supplied client-side)
+* Resource Type (supplied client-side)
+* Resource Version (supplied client-side)
+* MediaType: (e.g. `application/json`/`image`/`application/octet-stream`/`text/plain`) (computed ledger-side)
+* Created: XMLDatetime (computed ledger-side)
+* Checksum: SHA-256 (computed ledger-side)
+* previousVersionId: `null` if first, otherwise ID as long as Name, ResourceType, and MimeType match previous version (computed ledger-side)
+* nextVersionId: `null` if first/latest, otherwise ID as long as Name, ResourceType, and MimeType match previous version (computed ledger-side)
```

Comment:
`nextVersionId` should be `""` (an empty string) instead of null, so Resolver currently returns in this format.